meson.build

```diff
@@ -59,21 +59,35 @@ add_project_arguments(cc.get_supported_arguments(warning_flags), language: 'c')
 cdata = configuration_data()
 
 deps = []
+cc_args = []
 
 host_system = host_machine.system()
 
 if host_system == 'windows'
     deps += [cc.find_library('ws2_32')]
     deps += [cc.find_library('iphlpapi')]
+    building_for_vista = cc.compiles('''#include <windows.h>
+        #ifndef WINVER
+        #error "unknown minimum supported OS version"
+        #endif
+        #if (WINVER < _WIN32_WINNT_VISTA)
+        #error "Windows Vista API is not guaranteed"
+        #endif
+        ''',
+        name: 'building for Windows Vista')
+    if not building_for_vista
+        cc_args = [
+            '-D_WIN32_WINNT=_WIN32_WINNT_VISTA',
+            '-DWINVER=_WIN32_WINNT_VISTA',
```

Comment:
Don't use `_WIN32_WINNT_VISTA`, that define will not be available when building with older Windows SDKs. You should use the number directly: `0x0600`.

Reply:
If not defined it will error out, which is the whole point, this makes Vista the minimum required version, no?

Reply:
Ah, you're using failure to denote "not building for vista". Okay this is correct then.